pkg/controller/knativeserving/common/extensions.go

```diff
@@ -63,6 +64,14 @@ func (exts Extensions) Transform(instance *servingv1alpha1.KnativeServing) []mf.
 				UpdateConfigMap(u, data, log)
 			}
 		}
+		if u.GetKind() == "Deployment" {
```

Comment:
The comment on line 60 should get moved in to this function block now since it is not purely about config overriding

Reply:
Done

Reply:
I think it's one challenge here, just consider the image url in `deployment`. 
Actually, there is image url in `configmap`, event in `args` in `pod` definition.
Could we have a common solution to pick them up and do replacement, use `regex`  or something.

Reply:
I'm concerned with a common solution that supports replacing all possible values. I think if the configmap values need to be changed, the use should use the `config` key that mutates configmaps.

The `Image` kind needs to be able to be updated though.

Reply:
Updated to support `Image`